closed/src/java.base/share/classes/com/ibm/security/criu/NativePRNG.java

```diff
@@ -0,0 +1,248 @@
+     */
+    private static final class RandomIO {
+
+        // Holder for the seedFile.  Used if we ever add seed material.
+        private File seedFile;
+
+        // In/OutputStream for "seed" and "next".
+        private final InputStream seedIn, nextIn;
+        private OutputStream seedOut;
+
+        // flag indicating if we have tried to open seedOut yet
+        private boolean seedOutInitialized;
+
+        // mutex lock for generateSeed()
+        private final Object LOCK_GET_SEED = new Object();
+
+        // mutex lock for setSeed()
+        private final Object LOCK_SET_SEED = new Object();
+
+        // constructor, called only once from initIO()
+        private RandomIO(File seedFile, File nextFile) throws IOException {
+            this.seedFile = seedFile;
+            InputStream seedStream = null, nextStream = null;
+            try {
+                // Invoke the getInputStream method from the FileInputStreamPool class.
+                Class<?> runnable = Class.forName("sun.security.provider.FileInputStreamPool",
+                        true, ClassLoader.getSystemClassLoader());
+                Method getStream = runnable.getDeclaredMethod("getInputStream", File.class);
+                getStream.setAccessible(true);
+                seedStream = (InputStream) getStream.invoke(null, seedFile);
+                nextStream = (InputStream) getStream.invoke(null, nextFile);
+            } catch (Exception e) {
+                System.out.println(e.toString());
+            }
+            this.seedIn = seedStream;
+            this.nextIn = nextStream;
+        }
+
+        // read data.length bytes from in,
+        // These are not normal files, so we need to loop the read.
+        // just keep trying as long as we are making progress.
+        private static void readFully(InputStream in, byte[] data)
+                throws IOException {
+            int len = data.length;
+            int ofs = 0;
+            while (len > 0) {
+                int k = in.read(data, ofs, len);
+                if (k <= 0) {
+                    throw new EOFException("File(s) closed?");
+                }
+                ofs += k;
+                len -= k;
+            }
+            if (len > 0) {
+                throw new IOException("Could not read from file(s)");
+            }
+        }
+
+        // get true random bytes, just read from "seed"
+        private byte[] implGenerateSeed(int numBytes) {
+            synchronized (LOCK_GET_SEED) {
+                try {
+                    byte[] b = new byte[numBytes];
+                    readFully(seedIn, b);
+                    return b;
+                } catch (IOException e) {
+                    throw new ProviderException("generateSeed() failed", e);
+                }
+            }
+        }
+
+        // supply random bytes to the OS,
+        // write to "seed" if possible,
+        // always add the seed to our mixing random.
```

Comment:
Sentences should start with a capital, but this doesn't feel like it should be one sentence.

Reply:
changed to fragments